packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,523 @@
+                    // Show a message on this form.
+                    self.message = message.message
+                } else {
+                    // Handle moving to a new type of form.
+                    currentState = SignInController.shared!.currentRemediation!.type
+                }
+
+            case .failure(let error):
+                // Handle the error
+            }
+        }
+
+        let signInController = SignInController.shared?
+        if skip {
+            // Call a utility function for skipping the selection step.
+            signInController.skipAuthenticatorSelection(completion: completion)
+        } else {
+            // Select an authentictor and request the next step.
+            signInController.selectAuthenticator(authenticator: selectedOption,
+                                                 completion: completion)
+        }
+    }
+}
+
+```
+
+The UI updates the sign-on controller with the selected authenticator by calling `selectAuthenticator(authenticator:completion)`, or `skipAuthenticatorSelection(completion:)` if the user chose to skip enrolling in additional authenticators:
+
+```swift
+public func skipAuthenticatorSelection(completion:@escaping (Result<SuccessResultType, LoginError>) -> Void) {
+    // Set the current view model completion handler which is
+    // called when recieving a response from the SDK.
+    self.completion = completion
+
+    guard let skipRemediation = currentResponse?.remediations.filter({ $0.type == .skip }).first else {
+        // Handle the error
+        return
+    }
+
+    // Request the next step.
+    skipRemediation.proceed()
+}
+
+
+public func selectAuthenticator(authenticator: Remediation.Form.Field?, completion:@escaping (Result<SuccessResultType, LoginError>) -> Void) {
+    // Set the selected authenticator.
+    self.completion = completion
+
+    guard let selectorFields = authenticatorOptions() else {
+              self.finish(with: .cannotProceed)
+              return
+          }
+
+    // Update the remediation and send it to the server,
+    // requesting the next step.
+    selectorFields.optionSelectorField.selectedOption = authenticator
+    currentRemediation!.proceed()
+}
+```
+
+### 5: Display the shared secret, QR Code, and request the code
+
+After selecting the authenticator, the server sends an enrollment request in a remediation of type `.enrollAuthenticator`. Enrolling uses a secret key shared between Okta and Google Authenticator that's used to generate and verify the generated time-based one-time passcodes (TOTP).
+
+The remediation includes the shared secret as both a QR Code and as text. It also includes a request to enter a Google Authenticator code after enrollment.
+
+The sign-on controller code for `handleEnrollAuthenticator(remdiation:)` works for many different authenticators because the `form` property of the remediation contains the information required to build the UI:
```

Comment:
```suggestion
The sign-on controller code for `handleEnrollAuthenticator(remediation:)` works for many different authenticators because the `form` property of the remediation contains the information required to build the UI:
```

Also, perhaps this is worth emphasizing and putting it in a note. It's not text that's specific to the Google Auth guide.

Reply:
Fixed in text, code snippet, and source.

As to the note I'm not sure it's useful. There's plenty of Okta specific app code that's repeated in multiple flows, such as everything in steps 1 and 2.

Is there a reason to add the Note?

Reply:
Probably not. Just a thought.